lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2937,46 +2937,57 @@ namespace {
       if (!clangDecl->hasAttrs())
         return;
 
-      SmallVector<ValueDecl *, 1> results;
-      auto conformsToAttr =
-          llvm::find_if(clangDecl->getAttrs(), [](auto *attr) {
-            if (auto swiftAttr = dyn_cast<clang::SwiftAttrAttr>(attr))
-              return swiftAttr->getAttribute().starts_with("conforms_to:");
-            return false;
-          });
-      if (conformsToAttr == clangDecl->getAttrs().end())
+      SmallVector<clang::Attr*, 40> conformsToMatches;
+      llvm::copy_if(clangDecl->getAttrs(), std::back_inserter(conformsToMatches), [](auto *attr) -> bool {
+        if (auto swiftAttr = dyn_cast<clang::SwiftAttrAttr>(attr))
+          return swiftAttr->getAttribute().starts_with("conforms_to:");
+        return false; 
+      });
+
+      if (conformsToMatches.empty())
         return;
 
-      auto conformsToValue = cast<clang::SwiftAttrAttr>(*conformsToAttr)
-                                 ->getAttribute()
-                                 .drop_front(StringRef("conforms_to:").size())
-                                 .str();
+      for (auto match : conformsToMatches) {
+        if (auto swiftAttr = dyn_cast<clang::SwiftAttrAttr>(match))
```

Comment:
Do we ever expect this cast to fail? If we do not, I'd use `cast` instead of `dyn_cast` and remove the if statement.